src/rpc/mining.cpp

```diff
@@ -149,9 +152,50 @@ UniValue getnewblock(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_INVALID_PARAMS, "required_wait must be non-negative.");
     }
 
+    const std::string xfieldParam = request.params[2].isNull() ? "" : request.params[2].get_str();
+    TAPYRUS_XFIELDTYPES xfieldType = TAPYRUS_XFIELDTYPES::NONE;
+
+    //using lambda to avoid temp variables
+    xfieldType = [xfieldParam](int splitAt, TAPYRUS_XFIELDTYPES max) -> TAPYRUS_XFIELDTYPES
+        { int x = splitAt > 0 ? atoi(xfieldParam.substr(0,splitAt)) : 0;
+          return x > 0 && x < int(max) ? TAPYRUS_XFIELDTYPES(x) : TAPYRUS_XFIELDTYPES::NONE;
+        } (xfieldParam.find(':'), TAPYRUS_XFIELDTYPES::MAX_XFIELDTYPE );
+
+    if ( xfieldType ==  TAPYRUS_XFIELDTYPES::NONE )
+    {
+        throw JSONRPCError(RPC_INVALID_PARAMS, "xfield_type parameter could not be parsed. Check if the xfield parameter has format: <xfield_type:new_xfield_value>.");
+    }
+
```

Comment:
xfield type is optional. If unspecified, an empty xfield block must be created as before.

```
$ ./src/tapyrus-cli getnewblock 1AQ2CtG3jho78SrEzKe3vf6dxcEkJt5nzA                                          
error code: -32602
error message:
xfield_type parameter could not be parsed. Check if the xfield parameter has format: <xfield_type:new_xfield_value>.
```

Reply:
I didn't realise that this scenario was not tested! its covered now.

Reply:
Error occurs if only address and age are specified.

```
$ ./src/tapyrus-cli getnewblock 1Gx4oSoJjzhPnLTfCfobs52wU8LxVXpR2Z 0
error code: -1
error message:
JSON value is not an integer as expected
```